### PR TITLE
Improve MSRV CI check to print out problems to log

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

Closes https://github.com/apache/datafusion/issues/11788

## Rationale for this change
When the MSRV CI check fails it does not say what is wrong or offer specific advice on how to fix it. For example:

CI failure: https://github.com/apache/datafusion/actions/runs/10204347289/job/28280296775?pr=11627

Output error:
```
Run # If you encounter an error with any of the commands below
Fetching index
Verifying the Minimum Supported Rust Version (MSRV) for toolchain x86_64-unknown-linux-gnu
Using check command cargo check
   Failed check command cargo check didn't succeed
Crate source was found to be incompatible with its MSRV '1.76.0', as defined in '/__w/datafusion/datafusion/datafusion/core/Cargo.toml'
Error: Process completed with exit code 1.

```

## What changes are included in this PR?

1. Add `--logfile stdout` so errors are printed to the log
2. Update comments in CI check explaining how to fix. 

Here is an example failure with this change: https://github.com/apache/datafusion/actions/runs/10220955214/job/28282400260?pr=11789

While the log is not pretty 🤮  it at least provides enough information to debug:

> (/__w/datafusion/datafusion/datafusion/sql)\n    Checking datafusion-functions v40.0.0 (/__w/datafusion/datafusion/datafusion/functions)\nerror[E0716]: temporary value dropped while borrowed\n   --> datafusion/physical-expr-common/src/aggregate/groups_accumulator/prim_op.rs:167:18\n    |\n151 |         let values = match opt_filter {\n    |             ------ borrow later stored here\n...\n167 |                 &PrimitiveArray::<T>::new(values_buf, nulls_buf).with_data_type(dt)\n    |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ creates a temporary value which is freed while still in use\n168 |             }\n    |             - temporary value is freed at the end of this statement\n    |\n    = note: consider using a `let` binding to create a longer lived value\n\nFor more information about this error, try `rustc --explain E0716`.\nerror: could not compile `datafusion-physical-expr-common` (lib) due to 1 previous error\nwarning: build failed, waiting for other jobs to finish...\n" cmd="1.76.0-x86_64-unknown-linux-gnu cargo check"

## Are these changes tested?

Yes (in progress)

### CODE REVIEW

**Review by jayzhan211 — APPROVED** [review]
Great, I usually just binary search the failed version and run against it to get the compile error message, print out the log should be more helpful

**Comment by alamb** [comment]
> Great, I usually just binary search the failed version and run against it to get the compile error message, print out the log should be more helpful

😆  that is a good idea. I should have thought of that. 

I'll also add that as a comment to help others (any my future self)

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -566,18 +566,32 @@ jobs:
...
+          # 3. Update the MSRV version in `Cargo.toml`
+          #
+          # Please see the DataFusion Rust Version Compatibility Policy before
+          # updating Cargo.toml. You may have to update the code instead. 
+          # https://github.com/apache/datafusion/blob/main/README.md#rust-version-compatibility-policy
+          cargo msrv --verify --log-target stdout
       - name: Check datafusion-substrait
         working-directory: datafusion/substrait
-        run: cargo msrv verify
+        run: cargo msrv --verify --log-target stdout
```
Comment (korowa) [inline .github/workflows/rust.yml:591]: `--verify` flag has deprecation note in `cargo msrv --help`, so maybe it's better to continue using verify subcommand?

Comment (alamb) [reply .github/workflows/rust.yml:591]: Thank you -- this is a good tip. I added some more logging / output based on the docs in https://gribnau.dev/cargo-msrv/migration-guide/v0.15_v0.16_json.html

This command seems to add even more output
```shell
cargo msrv --output-format json --log-target stdout verify
```

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -566,18 +566,32 @@ jobs:
...
+          # 
+          # To resolve, either:
+          # 1. Change your code to use older Rust features, 
+          # 2. Revert dependency update
+          # 3. Update the MSRV version in `Cargo.toml`
+          #
+          # Please see the DataFusion Rust Version Compatibility Policy before
+          # updating Cargo.toml. You may have to update the code instead. 
+          # https://github.com/apache/datafusion/blob/main/README.md#rust-version-compatibility-policy
+          cargo msrv --verify --log-target stdout
```
Comment (korowa) [inline .github/workflows/rust.yml:588]:
```suggestion
          cargo msrv --output-format json --log-target stdout verify
```

**Review by korowa — APPROVED** [review]

File: .github/workflows/rust.yml (hunk 2 above)
Comment (alamb) [reply .github/workflows/rust.yml:588]: 🤦

**Comment by alamb** [comment]
Thanks again @korowa  and @jayzhan211

**Pull request merged**